tests/cleanup/remove_old_objects_openstack.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+# Script removes virtual machines and other artifacts older than HOURS_LIMIT (24 hours by default) from OpenStack
+
+. /usr/share/beakerlib/beakerlib.sh
+
+
+rlJournalStart
+    rlPhaseStartSetup
+        if [ -z "$OS_AUTH_URL" ]; then
+            rlFail "OS_AUTH_URL is empty!"
+        else
+            rlLogInfo "OS_AUTH_URL=$OS_AUTH_URL"
+        fi
+
+        if [ -z "$OS_USERNAME" ]; then
+            rlFail "OS_USERNAME is empty!"
+        else
+            rlLogInfo "OS_USERNAME is configured"
+        fi
+
+        if [ -z "$OS_PASSWORD" ]; then
+            rlFail "OS_PASSWORD is empty!"
+        else
+            rlLogInfo "OS_PASSWORD is configured"
+        fi
+
+        # VMs older than HOURS_LIMIT will be deleted
+        HOURS_LIMIT="${HOURS_LIMIT:-24}"
+        export TIMESTAMP=`date -u -d "$HOURS_LIMIT hours ago" '+%FT%T'`
+
+        rlLogInfo "HOURS_LIMIT=$HOURS_LIMIT"
+
+        for package in ansible python3-openstacksdk python3-openstackclient; do
+            if ! rlCheckRpm "$package"; then
+                rlRun -t -c "dnf -y install $package"
+                rlAssertRpm "$package"
+            fi
+        done
+
+        # update python3-osc-lib from testing-updates repo because of bug 1663043
+        rlRun -t -c "dnf -y --enablerepo=updates-testing update python3-osc-lib"
+
+        PLAYBOOK_DELETE_VMS=`mktemp`
+        PLAYBOOK_DELETE_IMAGES=`mktemp`
+    rlPhaseEnd
+
+    rlPhaseStartTest "Delete old VMs"
+        cat > $PLAYBOOK_DELETE_VMS << __EOF__
+- name: Delete old VMs
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Get VMs
+    os_server_facts:
+
+  - name: List all VMs
+    debug:
+      var: openstack_servers
+
+  - name: Delete old VMs
+    os_server:
+      name: "{{item.id}}"
+      state: absent
+    loop: "{{openstack_servers}}"
+    when: item.created < lookup('env','TIMESTAMP')
+    loop_control:
+      label: "{{item.name}} (id: {{item.id}} created: {{item.created}})"
+__EOF__
+
+        rlLogInfo "Removing VMs created before $TIMESTAMP"
+        rlRun -t -c "ansible-playbook $PLAYBOOK_DELETE_VMS"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Delete old images"
+        cat > $PLAYBOOK_DELETE_IMAGES << __EOF__
+- name: Delete old images
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Get images
+    os_image_facts:
+
+  - name: Delete old images
+    os_image:
+      name: "{{item.name}}"
+      id: "{{item.id}}"
+      state: absent
+    loop: "{{openstack_image}}"
```

Comment:
Question: how does the `openstack_image` variable work? Where does it come from ?